packages/@o3r/extractors/src/core/wrapper.ts

```diff
@@ -8,10 +13,59 @@ const noopBuilderWrapper: BuilderWrapper = (fn) => fn;
  * @param builderFn
  */
 export const createBuilderWithMetricsIfInstalled: BuilderWrapper = (builderFn) => async (opts, ctx) => {
+  const packageJsonPath = path.join(ctx.workspaceRoot, 'package.json');
+  const packageJson = existsSync(packageJsonPath)
+    ? JSON.parse(readFileSync(packageJsonPath, {encoding: 'utf8'}).toString())
+    : {};
   let wrapper: BuilderWrapper = noopBuilderWrapper;
   try {
     const { createBuilderWithMetrics } = await import('@o3r/telemetry');
-    wrapper = createBuilderWithMetrics;
-  } catch {}
+    if (packageJson.cli?.o3rMetrics) {
+      wrapper = createBuilderWithMetrics;
+    }
+  } catch (e: any) {
+    // Do not throw if `@o3r/telemetry is not installed
+    if (packageJson.cli?.o3rMetrics === true) {
+      ctx.logger.info('`cli.o3rMetrics` is set to true in your package.json, please install the telemetry package with `ng add @o3r/telemetry` to enable the collect of metrics.');
+    } else if ((!process.env.CI || process.env.CI === 'false') && typeof packageJson.cli?.o3rMetrics === 'undefined') {
+      ctx.logger.debug('`@o3r/telemetry` is not available.\nAsking to add the dependency\n' + e.toString());
+
+      const question: Question = {
+        type: 'confirm',
+        name: 'isReplyPositive',
+        message: `
+          Would you like to share anonymous usage data about this project with the Otter Team at Amadeus ?
+          It will help us to improve our tools.
+          For more details and instructions on how to change these settings, see https://github.com/AmadeusITGroup/otter/blob/main/docs/telemetry/README.md.
+        `
+      };
+      const { isReplyPositive } = await prompt([question]);
+
+      if (isReplyPositive) {
+        const pmr = getPackageManagerRunner(packageJson);
+
+        try {
+          const version = JSON.parse(readFileSync(path.join(__dirname, '..', '..', 'package.json'), 'utf-8')).version;
+          execFileSync(`${pmr} ng add @o3r/telemetry@${version}`);
+        } catch {
+          ctx.logger.warn('Failed to install `@o3r/telemetry`.');
+        }
+
+        const { createBuilderWithMetrics } = await import('@o3r/telemetry');
+        return createBuilderWithMetrics(builderFn)(opts, ctx);
+      } else {
+        ctx.logger.info('You can activate it at any time by running `ng add @o3r/telemetry`.');
+
+        packageJson.cli ||= {};
+        packageJson.cli.o3rMetrics = false;
+
+        if (existsSync(packageJsonPath)) {
+          writeFileSync(packageJsonPath, JSON.stringify(packageJson, null, 2));
```

Comment:
it's recommended to use `fs.promises` instead of sync functions as much as possible
```suggestion
          await fs.promises.writeFile(packageJsonPath, JSON.stringify(packageJson, null, 2));
```